common/djangoapps/student/tests/test_enrollment.py

```diff
@@ -73,7 +73,7 @@ def setUp(self):
         # We should also be enrolled as the honor mode.
         # Since honor and audit are currently offered together this precedence must
```

Comment:
This comment worries me.  The order you are changing in the auto_enroll_mode method above could have a greater impact.  We need to investigate the assumption in this comment.